src/sql/src/func.rs

```diff
@@ -2604,6 +2604,34 @@ lazy_static! {
                 // TODO(guswynn): The actual impl returns a i64, should this return that as well?
                 params![MapAny] => UnaryFunc::MapLength => Int32, oid::FUNC_MAP_LENGTH_OID;
             },
+            "headersmap" => Scalar {
```

Comment:
Could you please keep it under experimental for now?

Reply:
I did this, but I am not sure we have an easy way to add tests for experimental features!

Reply:
If I'm not mistaken, almost all of our tests run under experimental mode, including testdrive and SLTs.

Reply:
Oh i totally missed that!